packages/babel-plugin-display-name/src/index.js

```diff
@@ -239,6 +239,19 @@ function addDisplayNamesToFunctionComponent(t, path) {
 
     // const componentIdentifier = <Element />
     if (parentPath.isVariableDeclarator()) {
+      // Ternary expression
+      if (t.isConditionalExpression(parentPath.node.init)) {
+        const { consequent, alternate } = parentPath.node.init;
+        const isConsequentFunction =
+          t.isArrowFunctionExpression(consequent) || t.isFunctionExpression(consequent);
+        const isAlternateFunction =
+          t.isArrowFunctionExpression(alternate) || t.isFunctionExpression(alternate);
+
+        // Only add display name if variable is a function
+        if (!isConsequentFunction || !isAlternateFunction) {
+          return false;
+        }
+      }
```

Comment:
An alternative to this would be to make the `.displayName` assignment handle `null | undefined` values.